test/testchain/chain.go

```diff
@@ -0,0 +1,216 @@
+	// Create a new block packer with the current chain state and account information.
+	blkPacker := packer.New(c.Repo(), c.Stater(), account.Address, &genesis.DevAccounts()[0].Address, thor.NoFork)
+
+	// Schedule a new block to be packed at the current time.
+	blkFlow, err := blkPacker.Schedule(c.Repo().BestBlockSummary(), uint64(time.Now().Unix()))
+	if err != nil {
+		return fmt.Errorf("unable to schedule a new block: %w", err)
+	}
+
+	// Adopt the provided transactions into the block.
+	for _, trx := range transactions {
+		if err = blkFlow.Adopt(trx); err != nil {
+			return fmt.Errorf("unable to adopt tx into block: %w", err)
+		}
+	}
+
+	// Pack the adopted transactions into a block.
+	newBlk, stage, receipts, err := blkFlow.Pack(account.PrivateKey, 0, false)
+	if err != nil {
+		return fmt.Errorf("unable to pack tx: %w", err)
+	}
+
+	// Commit the new block to the chain's state.
+	if _, err := stage.Commit(); err != nil {
+		return fmt.Errorf("unable to commit tx: %w", err)
+	}
+
+	// Add the block to the repository.
+	if err := c.Repo().AddBlock(newBlk, receipts, 0); err != nil {
+		return fmt.Errorf("unable to add tx to repo: %w", err)
+	}
+
+	// Set the new block as the best (latest) block in the repository.
+	if err := c.Repo().SetBestBlockID(newBlk.Header().ID()); err != nil {
+		return fmt.Errorf("unable to set best block: %w", err)
+	}
+
+	return nil
+}
+
+// GetAllBlocks retrieves all blocks from the blockchain, starting from the best block and moving backward to the genesis block.
+// It limits the retrieval time to 5 seconds to avoid excessive delays.
+func (c *Chain) GetAllBlocks() ([]*block.Block, error) {
+	bestBlkSummary := c.Repo().BestBlockSummary()
+	var blks []*block.Block
+	currBlockID := bestBlkSummary.Header.ID()
+	startTime := time.Now()
+
+	// Traverse the chain backwards until the genesis block is reached or timeout occurs.
+	for {
+		blk, err := c.repo.GetBlock(currBlockID)
+		if err != nil {
+			return nil, err
+		}
+		blks = append(blks, blk)
+
+		// Stop when the genesis block is reached and reverse the slice to have genesis at position 0.
+		if blk.Header().Number() == c.genesisBlock.Header().Number() {
+			slices.Reverse(blks) // make sure genesis is at position 0
+			return blks, err
+		}
+		currBlockID = blk.Header().ParentID()
+
+		// Check if the retrieval process is taking too long (more than 5 seconds).
+		if time.Since(startTime) > 5*time.Second {
+			return nil, errors.New("taking more than 5 seconds to retrieve all blocks")
+		}
+	}
+}
+
+// BestBlock returns the current best (latest) block in the chain.
+func (c *Chain) BestBlock() (*block.Block, error) {
+	return c.Repo().GetBlock(c.Repo().BestBlockSummary().Header.ID())
+}
+
+// GetForkConfig returns the current fork configuration based on the ID of the genesis block.
+func (c *Chain) GetForkConfig() thor.ForkConfig {
+	return thor.GetForkConfig(c.GenesisBlock().Header().ID())
```

Comment:
`GetForkConfig` only returns the well know network config(mainnet and testnet). `forkconfig` should be a parameter that stored in struct `*chain`.

Reply:
Yeah, this is a fixed param after the `*chain` initialization.